### PR TITLE
config(datasets) - create a PluggableEntity class

### PR DESCRIPTION
**Pull request by onewland**
The goal of this new class is to provide a less flexible version of Entity that will not be subclassed but will be instantiated based on contents of YAML config files.

Once this and https://github.com/getsentry/snuba/pull/3046 are merged, we can make some changes to dynamically populate entities from YAML that fit certain characteristics (single storage plan, a limited set of query processors/mappers/etc.)

This should look somewhat similar to https://github.com/getsentry/snuba/pull/3037/files when done though that's obviously not complete

### CODE REVIEW

**Comment by volokluev** [comment]
question about this: the `get_entity` function returns an `Entity` class. How would these live side by side in the same process?

**Comment by onewland** [comment]
> question about this: the `get_entity` function returns an `Entity` class. How would these live side by side in the same process?

this returns an Entity instance, right?

```python
def get_entity(name: EntityKey) -> Entity:
    if name in ENTITY_IMPL:
        return ENTITY_IMPL[name]
```

**Comment by volokluev** [comment]
> something else would load up a separate map with YAML-loaded entities

yes so is that signature now going to be:

```python
def get_entity(name: EntityKey | str) -> Entity | PluggableEntity:
    pass
```

**Comment by onewland** [comment]
> > something else would load up a separate map with YAML-loaded entities
> 
> yes so is that signature now going to be:
> 
> ```python
> def get_entity(name: EntityKey | str) -> Entity | PluggableEntity:
>     pass
> ```

I was assuming we would keep that class return type as Entity (PluggableEntity is an Entity subclass)

**Comment by volokluev** [comment]
oh woops I missed that subclass. All good

File: tests/datasets/entities/test_pluggable_entity.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        original_body=query_body,
+        query=query,
+        snql_anonymized=snql_anonymized,
+        query_settings=HTTPQuerySettings(referrer=""),
+        attribution_info=AttributionInfo(get_app_id("blah"), "blah", None, None, None),
+    )
+    return request
+
+
+def test_generic_metrics_sets_vs_pluggable_similar_behavior(
```
Comment (volokluev) [inline tests/datasets/entities/test_pluggable_entity.py:124]:
```suggestion
def test_generic_metrics_sets_vs_pluggable_similar_pipeline_behavior(
```

**Review by volokluev — APPROVED** [review]

**Pull request merged**